src/plugins/controls/public/control_types/range_slider/range_slider_embeddable.tsx

```diff
@@ -307,12 +330,35 @@ export class RangeSliderEmbeddable extends Embeddable<RangeSliderEmbeddableInput
     rangeFilter.meta.type = 'range';
     rangeFilter.meta.params = params;
 
-    this.updateComponentState({ loading: false });
+    // Check if new range filter results in no data
+    if (!ignoreParentSettings?.ignoreValidations) {
+      filters.push(rangeFilter);
+
+      const timeFilter = this.dataService.timefilter.createFilter(dataView, timeRange);
+
+      if (timeFilter) {
+        filters.push(timeFilter);
+      }
+
+      const { min, max } = await this.fetchMinMax({ dataView, field, filters, query });
```

Comment:
Min and Max queries aren't extremely performance heavy, but we really only need a `doc_count` in order to tell if the potential filter set is valid or not. It seems to me that it might be slightly overkill, and a little developer experience unfriendly, to run the full `minMax` query here. I would run a really quick `size: 0` query with no `aggs` and just the query, then look at the `doc_count` returned in order to determine if the search is invalid or not.